gui/src/app/state/spend/step.rs

```diff
@@ -175,110 +188,121 @@ impl DefineSpend {
             }
         }
     }
-    fn auto_select_coins(&mut self, daemon: Arc<dyn Daemon + Sync + Send>) {
-        // Set non-input values in the same way as for user selection.
-        let mut outputs: HashMap<Address<address::NetworkUnchecked>, u64> = HashMap::new();
-        for recipient in &self.recipients {
-            outputs.insert(
-                Address::from_str(&recipient.address.value).expect("Checked before"),
-                recipient.amount().expect("Checked before"),
-            );
+    /// redraft calculates the amount left to select and auto selects coins
+    /// if the user did not select a coin manually
+    fn redraft(&mut self, daemon: Arc<dyn Daemon + Sync + Send>) {
+        if !self.form_values_are_valid() || self.recipients.is_empty() {
+            return;
         }
-        let feerate_vb = self.feerate.value.parse::<u64>().unwrap_or(0);
+
+        let destinations: Vec<(SpendOutputAddress, Amount)> = self
+            .recipients
+            .iter()
+            .map(|recipient| {
+                (
+                    SpendOutputAddress {
+                        addr: Address::from_str(&recipient.address.value)
+                            .expect("Checked before")
+                            .assume_checked(),
```

Comment:
I'm not sure if this matters when creating the spend, but `assume_checked` would set the network of a signet address to testnet. Following https://github.com/wizardsardine/liana/pull/813, we ignore the network when looking up addresses, but there might be some other effect of having a different network set.

Reply:
I do not think it matters, signet address are testnet address. https://github.com/wizardsardine/liana/blob/master/src/commands/mod.rs#L232 also address verification is done in the edit address message cycle and does not concern this PR.

Reply:
> Following https://github.com/wizardsardine/liana/pull/813, we ignore the network when looking up addresses

Since this is using the spend module it's not connected to the database anymore at all.

> but there might be some other effect of having a different network set.

As long as we are only using this dummy spend for selecting coins, the address encoding should have any effect. In fact we could be taking `Script`s instead of addresses in: https://github.com/wizardsardine/liana/blob/3e0f82a71e031604fcf8bb0fd757ae95908e8ca6/src/spend.rs#L385-L389

This would avoid requiring this kind of ugliness from the callers.